src/config.rs

```diff
@@ -22,19 +24,43 @@ pub enum ConfigurationError {
     MalformedSSORoleArn,
 }
 
-#[derive(Clone)]
+#[derive(Clone, Debug, PartialEq)]
 pub struct Credentials {
     pub region: Region,
     pub service_account_name: String,
-    pub role_arn: RoleArn,
+    pub role_arn: Option<RoleArn>,
+    pub aws_access_key_id: Option<String>,
+    pub aws_secret_access_key: Option<String>,
 }
 
 impl Credentials {
-    pub fn new(region: Region, service_account_name: String, role_arn: RoleArn) -> Self {
-        Self {
-            region,
-            service_account_name,
-            role_arn,
+    pub fn new(
+        region: Region,
+        service_account_name: String,
+        role_arn: Option<RoleArn>,
```

Comment:
_suggestion_: introduce an enum so you are sure either ARN or keys are set and you don't need to do the check below.

```
enum CredentialsMode {
   Role{arn: RoleArn},
   Keys{aws_access_key_id: String, aws_secret_access_key: String}
}

impl Credentials {
    pub fn new(
        region: Region,
        service_account_name: String,
        credentials_mode: CredentialsMode,) {}
 }
```